### PR TITLE
opendal: Bump to v0.0.5 to address performance regression and much UE enhancement

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR intends to adapt the opendal API changes, Fix https://github.com/datafuselabs/databend/issues/4197

More details could be found in https://github.com/datafuselabs/opendal/pull/35

## Changelog

- Performance Improvement

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by Xuanwo** [comment]
Our API looks much better! ping @sundy-li @dantengsky for benchmark test.

File: common/dal-context/src/context.rs [hunk 1]
```diff
@@ -22,9 +22,9 @@ use opendal::ops::OpStat;
 use opendal::ops::OpWrite;
 use opendal::readers::CallbackReader;
 use opendal::Accessor;
+use opendal::BoxedAsyncRead;
```
Comment (BohuTANG) [inline common/dal-context/src/context.rs:25]: LGTM.
A small question:
`BoxedAsyncRead` is `BoxedAsyncReader` or a `Read` opeartion?

Comment (Xuanwo) [reply common/dal-context/src/context.rs:25]: `BoxedAsyncRead` is the same with `Box<dyn AsyncRead>`

Comment (BohuTANG) [reply common/dal-context/src/context.rs:25]: Feel that `BoxedAsyncReader` is easier to understand as a reader, `BoxedAsyncRead` like a reading action :)

Comment (Xuanwo) [reply common/dal-context/src/context.rs:25]: LGTM, let's do this in opendal `v0.0.6`!

**Review by sundy-li — APPROVED** [review]

File: common/dal-context/src/context.rs (hunk 1 above)
Comment (Xuanwo) [reply common/dal-context/src/context.rs:25]: Tracked in https://github.com/datafuselabs/opendal/issues/52

**Comment by sundy-li** [comment]
Performance not expected, make this draft temporary.

**Comment by Xuanwo** [comment]
Merged with `main` to make sure we bench on the same base.

**Comment by BohuTANG** [comment]
Oops, has conflicts that must be resolved.

**Comment by Xuanwo** [comment]
> Oops, has conflicts that must be resolved.

Fixed.

**Review by sundy-li — APPROVED** [review]

**Comment by Xuanwo** [comment]
> The OpenDAL 0.5 looks not only related to the performance (I think the PR title misleading here), also improve error catch and some API design.

PR title updated :laughing:

**Pull request merged**